### ISSUE TITLE
vp9 in mp4 container?

### ISSUE THREAD

**Issue by xeno567** [yt-dlp/yt-dlp#7848]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### yt-dlp does not give out the required format:

The codec of the output video is now vp09, which it wasn’t before this issue occurred, when using this command:
`yt-dlp -f 'bv[height<=1080][ext=mp4]+ba[ext=m4a]/mp4' [YT-URL]`

vp09 can’t be played back on my device, I need H.264. 

I tried different configurations, downloading different videos, on their own, or in a playlist. Every time an unplayable vp09 video was the output. I received the following error messages:

```
WARNING: [youtube:tab] unable to extract yt initial data
WARNING: [youtube:tab] Incomplete yt initial data received.
WARNING: [youtube] Incomplete data received in embedded initial data; re-fetching using API.
```

Below, I inserted the verbose output from the video I originally wanted to download. (Beware, it's a k-pop video)

Please understand that I’m not really familiar or talented with coding or anything related, I’m just using yt-dlp for the long term reliability and I reported the issue because I was told to.
If you can offer me instant help by giving me another command, it would be greatly appreciated.

_I'm using yt-dlp for homebrew on Mac. Please do not refer me to other versions, as homebrew turned out to be the only viable method for me._

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '-f', 'bv[height<=1080][ext=mp4]+ba[ext=m4a]/mp4', 'https://www.youtube.com/watch?v=zi_6oaQyckM']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (pip)
[debug] Python 3.11.4 (CPython arm64 64bit) - macOS-13.4.1-arm64-arm-64bit (OpenSSL 3.1.2 1 Aug 2023)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.07.22, mutagen-1.46.0, sqlite3-2.6.0, websockets-11.0.3
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[youtube] Extracting URL: https://www.youtube.com/watch?v=zi_6oaQyckM
[youtube] zi_6oaQyckM: Downloading webpage
WARNING: [youtube] unable to extract initial player response; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[youtube] zi_6oaQyckM: Downloading ios player API JSON
[youtube] zi_6oaQyckM: Downloading android player API JSON
[youtube] zi_6oaQyckM: Downloading iframe API JS
[youtube] zi_6oaQyckM: Downloading player 3cd2d050
[youtube] zi_6oaQyckM: Downloading web player API JSON
[youtube] zi_6oaQyckM: Downloading m3u8 information
WARNING: [youtube] unable to extract yt initial data; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
WARNING: [youtube] Incomplete data received in embedded initial data; re-fetching using API.
[youtube] zi_6oaQyckM: Downloading initial data API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, vcodec:vp9.2, channels, acodec, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), channels, acodec, lang, proto, size, br, asr, vext, aext, hasaud, id
[info] zi_6oaQyckM: Downloading 1 format(s): 616+140
[debug] Invoking hlsnative downloader on "https://manifest.googlevideo.com/api/manifest/hls_playlist/expire/1692141129/ei/6bHbZMm4BYWi6dsP1Jy32AI/ip/2003:c9:af4e:dc00:1088:c64f:58d5:780b/id/ce2ffaa1a4327243/itag/616/source/youtube/requiressl/yes/ratebypass/yes/pfa/1/wft/1/sgovp/clen%3D71501643%3Bdur%3D208.416%3Bgir%3Dyes%3Bitag%3D356%3Blmt%3D1667368824399046/hls_chunk_host/rr2---sn-4g5ednsz.googlevideo.com/mh/YV/mm/31,26/mn/sn-4g5ednsz,sn-h0jeenle/ms/au,onr/mv/m/mvi/2/pl/42/initcwndbps/1775000/vprv/1/playlist_type/DVR/dover/13/txp/5532434/mt/1692119374/fvip/2/short_key/1/keepalive/yes/fexp/24007246/beids/24350018/sparams/expire,ei,ip,id,itag,source,requiressl,ratebypass,pfa,wft,sgovp,vprv,playlist_type/sig/AOq0QJ8wRgIhAO5KPIrUbXyVnl-GVoVWoBSJwpIAqbHeXahpdxEQwDQYAiEAovNAxU2-1h6gXAhQYvAcaqMOkX0Ip-X12OoiOTgTB7k%3D/lsparams/hls_chunk_host,mh,mm,mn,ms,mv,mvi,pl,initcwndbps/lsig/AG3C_xAwRAIgOGwGluHUqmgL9_jASu5XSGeXyUiTm6KmhQE11jV2iSwCIBh8rGt91t2J5eToOho1ssWbzVmtu-9PnvgkYdaAsA-z/playlist/index.m3u8"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 41
[download] Destination: TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].f616.mp4
[download] 100% of   68.21MiB in 00:01:11 at 975.37KiB/s
[debug] Invoking http downloader on "https://rr2---sn-4g5ednsz.googlevideo.com/videoplayback?expire=1692141129&ei=6bHbZMm4BYWi6dsP1Jy32AI&ip=2003%3Ac9%3Aaf4e%3Adc00%3A1088%3Ac64f%3A58d5%3A780b&id=o-AAHsWLjg9rRBtgW7hkSVzc34yhmjE4RVdlUs61dCIdBd&itag=140&source=youtube&requiressl=yes&mh=YV&mm=31%2C26&mn=sn-4g5ednsz%2Csn-h0jeenle&ms=au%2Conr&mv=m&mvi=2&pl=42&initcwndbps=1775000&vprv=1&svpuc=1&mime=audio%2Fmp4&gir=yes&clen=3374512&dur=208.468&lmt=1575018300672107&mt=1692119374&fvip=2&keepalive=yes&fexp=24007246&beids=24350018&c=IOS&txp=5531432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Csvpuc%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRgIhAIirlFRD0C_y3nXi90eX5XH0ULgZac25wBxgpr7e8Nt6AiEAi304r0PjBAX50DZBOo3xeoEdxJfylLb724OxInjsy2U%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhALh1MgYWyIB04bgN9wCJs49Tyo6EAW44MfFP7zQsTGTcAiEA_RqtjHvNiGregH_HzqDruetOXnM2-DXuprcFdVut_bM%3D"
[download] Destination: TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].f140.m4a
[download] 100% of    3.22MiB in 00:00:03 at 1.02MiB/s
[Merger] Merging formats into "TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].f616.mp4' -i 'file:TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].f140.m4a' -c copy -map 0:v:0 -map 1:a:0 -movflags +faststart 'file:TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].temp.mp4'
Deleting original file TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].f616.mp4 (pass -k to keep)
Deleting original file TWICE ＂Merry & Happy＂ M⧸V [zi_6oaQyckM].f140.m4a (pass -k to keep)
```

**Comment by tzvc** [comment]
Running into the same issue with the following params: `bestvideo[height<=1080][ext=mp4]+bestaudio[ext=m4a]/best[height<=1080][ext=mp4]`

**Comment by bashonly** [comment]
The different codec is unrelated to the "unable to extract yt intiial data" issue (which was #7594 and fixed by 378ae9f9fb8e8c86e6ac89c4c5b815b48ce93620, install master branch if you want the patch)

Format 616 is vp9 in mp4 container. That's just how Youtube serves it. So it is not "wrong," it is just not what you were expecting.

If you don't want vp9, then explicitly filter it out with `[vcodec!~='vp0?9']`

mp4 is just a container. Using `ext=mp4` does not guarantee any particular codec. But if you want h264/avc (which is what most people seem to want when they say they want mp4), you could do something like this instead of your `-f` arg:
`-S vcodec:h264,res,acodec:m4a`
